tests/core/engine_adapter/integration/test_integration.py

```diff
@@ -1731,6 +1733,8 @@ def _normalize_snowflake(name: str, prefix_regex: str = "(sqlmesh__)(.*)"):
     if config.model_defaults.dialect != ctx.dialect:
         config.model_defaults = config.model_defaults.copy(update={"dialect": ctx.dialect})
 
+    # To enable parallelism in integration tests
```

Comment:
Why does this impact parallelism?

Reply:
To isolate the connections per test because otherwise they will encounter concurrency issues, as they will end up using the same connections. Since all the engine adapters are created upfront and this particular file's tests are using for the context the same config: https://github.com/TobikoData/sqlmesh/blob/main/tests/core/engine_adapter/integration/config.yaml